libs/books/feature/src/lib/book-search/book-search.component.ts

```diff
@@ -51,10 +57,21 @@ export class BookSearchComponent {
   }
 
   searchBooks() {
-    if (this.searchForm.value.term) {
-      this.store.dispatch(searchBooks({ term: this.searchTerm }));
-    } else {
-      this.store.dispatch(clearSearch());
-    }
+    // get search value from form
+    this.searchForm.valueChanges.pipe(
+        takeUntil(this.unsubscribe$),
+        debounceTime(500),
+        distinctUntilChanged()
```

Comment:
Are these the right way round? What would change if you put them the other way round?

Reply:
moved distinctUntilChanged() to top and updated takeUntil as the last operator.